internal/fnruntime/utils.go

```diff
@@ -140,17 +140,33 @@ type SelectionContext struct {
 }
 
 // SelectInput returns the selected resources based on criteria in selectors
-func SelectInput(input []*yaml.RNode, selectors []kptfilev1.Selector, _ *SelectionContext) ([]*yaml.RNode, error) {
+func SelectInput(input []*yaml.RNode, selectors, exclusions []kptfilev1.Selector, _ *SelectionContext) ([]*yaml.RNode, error) {
+	var selectedInput []*yaml.RNode
 	if len(selectors) == 0 {
-		return input, nil
+		selectedInput = input
+	} else {
+		for _, node := range input {
+			for _, selector := range selectors {
+				if isMatch(node, selector) {
+					selectedInput = append(selectedInput, node)
+				}
+			}
+		}
+	}
+	if len(exclusions) == 0 {
+		return selectedInput, nil
 	}
 	var filteredInput []*yaml.RNode
-	for _, node := range input {
-		for _, selector := range selectors {
-			if isMatch(node, selector) {
-				filteredInput = append(filteredInput, node)
+	for _, node := range selectedInput {
+		matchesExclusion := false
+		for _, exclusion := range exclusions {
+			if !exclusion.IsEmpty() && isMatch(node, exclusion) {
+				matchesExclusion = true
 			}
 		}
+		if !matchesExclusion {
+			filteredInput = append(filteredInput, node)
+		}
```

Comment:
this can probably be moved in the block where we mark `matchesExclusion = true` and break  and get rid of `matchesExclusion` ?

Reply:
I don't think we can move it in, because we need to do this if it doesn't match any of the exclusions (we are checking for `!matchesExclusion` here. So we need to check all exclusions before we can decide whether or not we can include this input.

Reply:
Aah...yep!